lib/reek/context_builder.rb

```diff
@@ -133,20 +148,12 @@ def process_defs(exp)
     # we also record to what the method call is referring to
     # which we later use for smell detectors like FeatureEnvy.
     #
-    # :reek:TooManyStatements: { max_statements: 7 }
-    # :reek:FeatureEnvy
     def process_send(exp)
-      method_name = exp.method_name
-      if exp.visibility_modifier?
-        element.track_visibility(method_name, exp.arg_names)
-      elsif exp.attribute_writer?
-        exp.args.each do |arg|
-          append_new_context(Context::AttributeContext, arg, exp)
-        end
-      else
-        append_new_context(Context::SendContext, exp, method_name)
+      if current_context.module_context?
```

Comment:
I think this began when you created two different methods with name `handle_` and including the type in the method name. If you can find a better naming for what these two methods do, you could either find a single abstraction that fits for both contexts, or you can find two different methods that apply in both contexts, but maybe with no effects for some invocations.